### PR TITLE
auth0 integration + google sign in refactor + user model (compatible with both auth0 and google signin)

### PR DESCRIPTION
**Pull request by ViAsmit**
Closes: #7

### CODE REVIEW

**Comment by ViAsmit** [comment]
@jddeep @chandansgowda  please go through it once...
Task done:

1. Moved Chandan's code to combined page both auth0 and google login
2. integrated auth0 login with email
3. added user model, compatible with both auth0 user, and google user

Video Attached:
 (Since its iOS, google sign in without webview is still not possible because its iphone is not a google product
 but in android, it work as same shown in @chandansgowda PR video)
 
 

https://user-images.githubusercontent.com/55342789/229219880-c5e99ae8-f1ee-42e8-9e47-9e534b85a144.mov

File: lib/controllers/profile_controller.dart [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+
+class ProfileController extends GetxController{
+  bool isLoading = false;
+  User userProfile = Get.arguments[0];
+
+  @override
+  void onInit() {
+    super.onInit();
+    print("User");
```
Comment (jddeep) [inline lib/controllers/profile_controller.dart:15]: Please avoid using `print`. Use `log` instead.

Comment (ViAsmit) [reply lib/controllers/profile_controller.dart]: Done!

File: lib/controllers/authentication_controller.dart [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  Future<void> loginWithGoogle() async {
+    try {
+      GoogleSignInAccount? googleUser = await googleSignIn.signIn();
+      print(googleUser?.displayName!);
+      user = User(
+        name: googleUser?.displayName,
+        email: googleUser?.email,
+        pictureUrl: googleUser?.photoUrl,
+        signedInBy: SignedInBy.google,
+      );
```
[the remark this answers is not in the document]
Comment (ViAsmit) [reply lib/controllers/authentication_controller.dart]: we can discuss it over a call... let me know when you are available

[the remark this answers is not in the document]
Comment (chandansgowda) [reply lib/controllers/authentication_controller.dart]: @ViAsmit @jddeep As far as my research goes, we just have one workaround here. It is to create an account on Auth0 as soon as the user signs in through google using their email address and a default strong password and use that account itself for the app. So, everytime the user signs in using google, we will use their Auth0 instance. The problem with this approach is that if someone comes to know about the default password, he can gain access to all the accounts using Google SignIn. So, what if we create a collection for users on Firebase and store their details over there?

**Review by jddeep — APPROVED** [review]

**Pull request merged**